modules/core/src/persistence_base64.cpp

```diff
@@ -803,9 +811,28 @@ class BinaryToCvSeqConvertor
                 }; // need a better way for outputting error.
 
                 offset = static_cast<size_t>(cvAlign(static_cast<int>(offset), static_cast<int>(size)));
-                pack.offset = offset;
+                if (offset != offset_packed)
+                {
+                    static bool skip_message = cv::utils::getConfigurationParameterBool("OPENCV_PERSISTENCE_SKIP_PACKED_STRUCT_WARNING",
+#ifdef _DEBUG
+                            false
+#else
+                            true
+#endif
+                    );
+                    if (!skip_message)
+                    {
+                        CV_LOG_WARNING(NULL, "Binary converter: struct storage layout has been changed in OpenCV 3.4.7. Alignment gaps has been removed from the storage containers. "
```

Comment:
Can we use `CV_LOG_DEBUG` and avoid introduction of new environment configuration parameter?

Reply:
`DEBUG` messages are not compiled into release builds by default